pkg/server/plugin/keymanager/gcpkms/gcpkms.go

```diff
@@ -1103,8 +1117,14 @@ func parseAndValidateConfig(c string) (*Config, error) {
 		return nil, status.Error(codes.InvalidArgument, "configuration is missing the key ring")
 	}
 
-	if config.KeyMetadataFile == "" {
-		return nil, status.Error(codes.InvalidArgument, "configuration is missing server ID file path")
+	if config.KeyMetadataFile == "" && config.KeyIdentifierFile == "" && config.KeyIdentifierValue == "" {
```

Comment:
In the case of the Google Cloud Key Management Service, we leverage the use of labels to identify which keys belongs to a server. We should validate the configured `key_identifier_value` according with the requirements for the label values (63 chars length and other restrictions): https://cloud.google.com/kms/docs/creating-managing-labels#requirements

Reply:
The current implementation requires a UUID in the file. Are we saying we'd allow non-UUIDs in that file to match the verification behavior you've linked?

https://github.com/spiffe/spire/blob/f8f6c0e46d67b99ee4c91afe80766699c4bc063b/pkg/server/plugin/keymanager/gcpkms/gcpkms.go#L1024

Reply:
I changed the value validation. But probably should consider whether the file behavior should match.

Reply:
>Are we saying we'd allow non-UUIDs in that file to match the verification behavior you've linked?

No, I think that the content of the file should remain to be opaque to the user.
The user is not meant to manipulate the content of the file configured in key_identifier_file (key_metadata_file), only its path/name. I think that the behavior should remain the same in case that a key_identifier_file is configured. The logic here is "let SPIRE manage the key_identifier_value".

In the case that a key_identifier_value is configured, I personally think that having a more relaxed format there (following the restrictions imposed by the service) would be useful, so a value that's meaningful for the environment where SPIRE Server is running can be used, probably expanding environment variables. This would be particularly useful in the k8s case.

Reply:
OK.